app/src/main/java/org/xbmc/kore/ui/sections/localfile/LocalFileListFragment.java

```diff
@@ -0,0 +1,110 @@
+        directoryMusicFileListArgs.putParcelable(LocalPictureListFragment.SORT_METHOD, sortMethod);
+
+        Bundle directoryMoviesFileListArgs = new Bundle();
+        String directoryMovies = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_MOVIES).getAbsolutePath();
+        directoryMoviesFileListArgs.putString(LocalPictureListFragment.ROOT_PATH_LOCATION, directoryMovies);
+        directoryMoviesFileListArgs.putParcelable(LocalPictureListFragment.SORT_METHOD, sortMethod);
+
+        Bundle externalStorageFileListArgs = new Bundle();
+        String externalStorage = Environment.getExternalStorageDirectory().getAbsolutePath();
+        externalStorageFileListArgs.putString(LocalPictureListFragment.ROOT_PATH_LOCATION, externalStorage);
+        externalStorageFileListArgs.putParcelable(LocalPictureListFragment.SORT_METHOD, sortMethod);
+
+        TabsAdapter tabsAdapter = new TabsAdapter(getActivity(), getChildFragmentManager())
+                .addTab(LocalPictureListFragment.class, dcimFileListArgs, R.string.dcim, 1)
+                .addTab(LocalPictureListFragment.class, directoryMusicFileListArgs, "Music", 2)
+                .addTab(LocalPictureListFragment.class, directoryMoviesFileListArgs, "Movies", 3)
+                .addTab(LocalPictureListFragment.class, externalStorageFileListArgs, R.string.external_storage, 4);
+        Environment.getRootDirectory();
+        File[] externalFilesDirs = ContextCompat.getExternalFilesDirs(getActivity(),null);
+        for (int i = 0; i < externalFilesDirs.length; i++) {
+            File file = externalFilesDirs[i].getParentFile().getParentFile().getParentFile().getParentFile();
+
+            if (file.getAbsolutePath().equals(externalStorage))
+                continue;
+            Bundle bundle = new Bundle();
+            bundle.putString(LocalPictureListFragment.ROOT_PATH_LOCATION, file.getAbsolutePath());
+            bundle.putParcelable(LocalPictureListFragment.SORT_METHOD, sortMethod);
+
+            tabsAdapter.addTab(LocalPictureListFragment.class, bundle, file.getName(),i+2);
+        }
+
+        return tabsAdapter;
+    }
+
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+        try {
+            LocalFileActivity listenerActivity = (LocalFileActivity) activity;
+            listenerActivity.setBackPressedListener(this);
+        } catch (ClassCastException e) {
+            throw new ClassCastException(activity.toString() + " unable to register BackPressedListener");
+        }
+    }
+
+    @Override
+    public boolean onBackPressed() {
+        // Tell current fragment to move up one directory, if possible
+        LocalPictureListFragment curPage = (LocalPictureListFragment)((TabsAdapter)getViewPager().getAdapter())
+                .getStoredFragment(getViewPager().getCurrentItem());
+        if ((curPage != null) && !curPage.atRootDirectory()) {
```

Comment:
It seems the `atRootDirectory` check is faulty. I get into a loop when pressing back on the music tab (currently only one I'm trying to test).

Reply:
This particular code is equal to the existing code in the `Files` section, so maybe we have a bug there that needs solving.